httpie/codec.py

```diff
@@ -0,0 +1,77 @@
+from argparse import Namespace
+from typing import Union
+
+from requests.utils import get_encoding_from_headers
+
+from .cli.constants import DEFAULT_FORMAT_BODY_CHARSET
+from .constants import UTF8
+from .context import Environment
+
+
+def detect(content: bytes) -> str:
+    """Detect the charset based on `content`.
+    Fallback to UTF-8 when no suitable encoding found.
+
+    """
+    from charset_normalizer import from_bytes
+
+    match = from_bytes(bytes(content)).best()
+    return match.encoding if match else UTF8
+
+
+class Codec:
```

Comment:
In my opinion, it is a bit misleading. `Codec` does not reflect what you intend to do here.
Perhaps `TextDecoderStrategy` instead?

Reply:
Indeed, it seems more appropriate :+1: